### PR TITLE
Preparing release 0.5.0

### PR DESCRIPTION
**Pull request by mdouaihy**
Signed-off-by: Mehrez DOUAIHY <mehrez.douaihy@gmail.com>

## Which problem is this PR solving?
- Prepare for release 0.4.3

## Short description of the changes
- Prepare the changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
-- Nothing yet
+- Improve C++ flag handling in CMake (#128)
+- localIP: try resolving local hostname first (#130)
+- Fix deadlock if steady_clock::now() returns the same value twice (#132)
+- Implement SpanContext::Clone (#138)
+- Change sampler param sentinel value from YAML parser (#145)
+- Fix RemoteReporter test race condition (#135)
+- Add details on how to set agent & collector endpoint (#153)
+- Clarify agent/sampler address overrides in README
+- Upgrade to OpenTracing 1.5.0; Support build on Windows (#115)
```
Comment (yurishkuro) [inline CHANGELOG.md]: this should certainly deserve a bump to 0.5.0, not a patch

Comment (mdouaihy) [reply CHANGELOG.md]: indeed, I hesitated bumping the version between choosing a name for the branch and editing the changelog.

Comment (mdouaihy) [reply CHANGELOG.md]: change pushed.

**Comment by yurishkuro** [comment]
What about the other change `project(jaegertracing VERSION 0.5.0)`?

Usually we have all clients include their versions as span.process.tags, but it's more difficult with clients that are only released via tag (like Go), so in those we usually have a constant that needs to be bumped as part of the release.

**Comment by mdouaihy** [comment]
This is done automatically at build time via CMake

```
static constexpr auto kJaegerClientVersion = "C++-@PROJECT_VERSION@";
```

**Comment by yurishkuro** [comment]
0.5.0 released

**Pull request merged**